### DISCUSSION TITLE
Is possible to intercept the "save" button to send the PDF to another application?

### DISCUSSION THREAD

**Discussion by monkiki** [mozilla/pdf.js#17348] [General]
I would like to know if there is any way to intercept the "save" toolbar button so the annotated PDF can be summited to another URL.

Thanks in advance.

**Comment by lasseborly** [comment]
Could you elaborate a bit @monkiki ?
You can change the name of the saved file from the url by manipulating the `PDFViewerApplication._contentDispositionFilename`.

`PDFViewerApplication._contentDispositionFilename = "some_other_filename.pdf"`

**Comment by monkiki** [comment]
Sorry, I wanted to know if I can configure an HTTP POST endpoint where pdf.js would send the modified PDF and not just download it.

Thanks in advance.

**Reply by lasseborly** [reply]
No, I do not believe so. I think you would need to fork the viewer code and do that implementation yourself in the downloadManager.download function.

Or if you really want to avoid forking you could possibly remove the eventlisteners from the button, add your own listener and re-implement the functionality that you would find in the `save` function. That is a lot of code.

Sorry. There does not seem to be any hooks that would allow for what you are wanting.

**Reply by monkiki** [reply]
Thanks for the tips!

**Comment by lasseborly** [comment]
It actually wasn't that bad @monkiki 

```javascript
PDFViewerApplication.save = async () => {
      if (PDFViewerApplication._saveInProgress) {
        return;
      }
      PDFViewerApplication._saveInProgress = true;
      await PDFViewerApplication.pdfScriptingManager.dispatchWillSave();
      const url = PDFViewerApplication._downloadUrl;
      const filename = PDFViewerApplication._docFilename;
      try {
        PDFViewerApplication._ensureDownloadComplete();
        const data = await PDFViewerApplication.pdfDocument.saveDocument();
        const blob = new Blob([data], {
          type: "application/pdf",
        });
        console.log(blob, data, url, filename);
        // await PDFViewerApplication.downloadManager.download(
        //   blob,
        //   url,
        //   filename,
        //   options
        // );
      } catch (reason) {
        console.error(`Error when saving the document: ${reason.message}`);
        await PDFViewerApplication.download(options);
      } finally {
        await PDFViewerApplication.pdfScriptingManager.dispatchDidSave();
        PDFViewerApplication._saveInProgress = false;
      }
    };
```

Then just upload the to whatever endpoint you desire.

**Reply by monkiki** [reply]
Great! Thanks a lot!

**Reply by Elesoki** [reply]
This is a great idea, how can it be use to save/overwrite back onto the original PDF file on the server?
